packages/runtime/src/lib/txn.ts

```diff
@@ -119,3 +119,106 @@ export function txAppArg (txField: TxField, tx: Txn, idx: number, op: Op,
+ For more advanced use-cases, please use `algosdk.tx` directly.
+ NOTE: parseSSCAppArgs is used to handle case when user passes appArgs similar to goal
+ * @param execParams ExecParams
+ * @param suggestedParams Suggested params
+ * @returns SDK Transaction object
+ */
+export function mkTransaction (execParams: ExecParams, suggestedParams: SuggestedParams): Transaction {
+  const note = encodeNote(execParams.payFlags.note, execParams.payFlags.noteb64);
+  const transactionType = execParams.type;
+  switch (execParams.type) {
+    case TransactionType.TransferAsset: {
+      return algosdk.makeAssetTransferTxnWithSuggestedParams(
+        execParams.fromAccount.addr,
+        execParams.toAccountAddr,
+        execParams.payFlags.closeRemainderTo,
+        undefined,
+        execParams.amount,
+        note,
+        execParams.assetID,
+        suggestedParams);
+    }
+    case TransactionType.TransferAlgo: {
+      return algosdk.makePaymentTxnWithSuggestedParams(
+        execParams.fromAccount.addr,
+        execParams.toAccountAddr,
+        execParams.amountMicroAlgos,
+        execParams.payFlags.closeRemainderTo,
+        note,
+        suggestedParams);
+    }
+    case TransactionType.ClearSSC: {
+      return algosdk.makeApplicationClearStateTxn(
+        execParams.fromAccount.addr,
+        suggestedParams,
+        execParams.appId,
+        parseSSCAppArgs(execParams.appArgs),
+        execParams.accounts,
+        execParams.foreignApps,
+        execParams.foreignAssets,
+        note,
+        execParams.lease,
+        execParams.rekeyTo
+      );
+    }
+    case TransactionType.DeleteSSC: {
+      return algosdk.makeApplicationDeleteTxn(
+        execParams.fromAccount.addr,
+        suggestedParams,
+        execParams.appId,
+        parseSSCAppArgs(execParams.appArgs),
+        execParams.accounts,
+        execParams.foreignApps,
+        execParams.foreignAssets,
+        note,
+        execParams.lease,
+        execParams.rekeyTo
+      );
+    }
+    case TransactionType.CallNoOpSSC: {
+      return algosdk.makeApplicationNoOpTxn(
+        execParams.fromAccount.addr,
+        suggestedParams,
+        execParams.appId,
+        parseSSCAppArgs(execParams.appArgs),
+        execParams.accounts,
+        execParams.foreignApps,
+        execParams.foreignAssets,
+        note,
+        execParams.lease,
+        execParams.rekeyTo);
+    }
+    case TransactionType.CloseSSC: {
+      return algosdk.makeApplicationCloseOutTxn(
+        execParams.fromAccount.addr,
+        suggestedParams,
+        execParams.appId,
+        parseSSCAppArgs(execParams.appArgs),
+        execParams.accounts,
+        execParams.foreignApps,
+        execParams.foreignAssets,
+        note,
+        execParams.lease,
+        execParams.rekeyTo
+      );
+    }
+    default: {
+      throw new TealError(ERRORS.TEAL.UNKNOWN_TRANSACTION_TYPE, { transaction: transactionType });
```

Comment:
Isn't `transactionType` a number here? (`execParams.type` is an enum)

Reply:
i will check, i just moved it here, is everything OK otherthan this @ratik21  ?

Reply:
@amityadav0 looks good. Thanks for the PR.

Reply:
i think this is not a problem just confirmed by running,
this shows string if we pass unknown because we are storing this before switch.